src/libtest/lib.rs

```diff
@@ -514,6 +723,42 @@ fn is_nightly() -> bool {
     bootstrap || !disable_unstable_features
 }
 
+// Gets the option value and checks if unstable features are enabled.
+macro_rules! unstable_optflag {
+    ($matches:ident, $allow_unstable:ident, $option_name:literal) => {{
+        let opt = $matches.opt_present($option_name);
+        if !$allow_unstable && opt {
+            return Some(Err(format!(
+                "The \"{}\" flag is only accepted on the nightly compiler",
+                $option_name
+            )));
+        }
+
+        opt
+    }};
+}
+
+// Gets the CLI options assotiated with `report-time` feature.
+fn get_time_options(
+    matches: &getopts::Matches,
+    allow_unstable: bool)
+-> Option<OptPartRes<TestTimeOptions>> {
```

Comment:
Talking about quite over-complicated return type: as I mentioned before, I want to split `libtest` into several modules, so every module will be able to parse options of itself.
Thus, we will have a bunch of functions like `get_XXX_options` which will return `Option<OptPartRes<T>>`, and one master function in `cli.rs` which will combain options parts from different modules.

However, as I can see, currently `None` is returned only if `help` argument was passed, so I guess I'll be able to simplify signature to be just `OptPartRes<T>`.